src/main/java/io/fabric8/elasticsearch/plugin/acl/DynamicACLFilter.java

```diff
@@ -238,98 +274,135 @@ private synchronized void syncAcl() {
-			.setSource(mapper.writeValueAsBytes(acl))
-			.setRefresh(true).request();
-		request.putInContext(OS_ES_REQ_ID, ACL_FILTER_ID);
-		esClient.index(request).actionGet();
-		lock.lock();
-		try{
-			logger.debug("Waiting up to {} ms. to be notified that SearchGuard has refreshed the ACLs", aclSyncDelay);
-			syncing.await(aclSyncDelay, TimeUnit.MILLISECONDS);
-		}catch(InterruptedException e){
-			logger.error("Error while awaiting notification of ACL load by SearchGuard", e);
-		}finally{
-			lock.unlock();
-		}
+	private void writeACL(Client esClient, SearchGuardRoles roles, SearchGuardRolesMapping rolesMapping) throws IOException {
+		
+		IndexRequest rolesIR = new IndexRequest("searchguard").type("roles").id("0").refresh(true)
+        .consistencyLevel(WriteConsistencyLevel.DEFAULT)
+        .source(roles.toMap());
+		
+		String rID = esClient.index(rolesIR).actionGet().getId();
+		logger.debug("Built roles request: '{}'", rolesIR);
+		logger.debug("Roles ID: '{}'", rID);
+
+		IndexRequest mappingIR = new IndexRequest("searchguard").type("rolesmapping").id("0").refresh(true)
+                .consistencyLevel(WriteConsistencyLevel.DEFAULT)
+                .source(rolesMapping.toMap());
+		
+		String rmID = esClient.index(mappingIR).actionGet().getId();
+		logger.debug("Built rolesMapping request: '{}'", mappingIR);
+		logger.debug("rolesMapping ID: '{}'", rmID);
+
+		// force a config reload
+		String[] updateString = {"roles", "rolesmapping"};		
+		ConfigUpdateResponse cur = esClient.execute(ConfigUpdateAction.INSTANCE, new ConfigUpdateRequest(updateString)).actionGet();
+		
+		if ( cur.getNodes().length > 0 )
+			logger.debug("Successfully reloaded config with '{}' nodes", cur.getNodes().length);
+		else
+			logger.warn("Failed to reloaded config -- '{}' nodes updated", cur.getNodes().length);
 	}
 	
-	private void seedInitialACL(Client esClient) throws Exception {
+	private void createConfig(Client esClient) throws IOException {
+		
+		final ClusterHealthResponse chr = esClient.admin().cluster().health(new ClusterHealthRequest().waitForYellowStatus()).actionGet();
 
-		SearchGuardACL acl = new SearchGuardACL();
-		boolean create = false;
+        if ( ! chr.isTimedOut() ) {
+		
+			esClient.admin().indices().create( new CreateIndexRequest().index(searchGuardIndex)
+					.settings("index.number_of_shards", 1, "index.number_of_replicas", chr.getNumberOfDataNodes()-1)
+	                ).actionGet().isAcknowledged();
+
+			// Wait for health status of YELLOW
+			ClusterHealthRequest healthRequest = new ClusterHealthRequest()
+					.indices(new String[]{searchGuardIndex}).waitForYellowStatus();
+			
+			esClient.admin().cluster().health(healthRequest).actionGet().getStatus();
+			
+			String cd = settings.get(SG_CONFIG_SETTING_PATH, DEFAULT_SG_CONFIG_SETTING_PATH);
+			// create initial sg config
+			boolean success = uploadFile(esClient, cd+"sg_config.yml", "config");
```

Comment:
There seems to be a loop here that could be more elegant.  Additionally, it seems there is possibility that we have partial success.  What are the ramifications of not uploading all the files?  Also, if we were able to either loop or return something other then boolean, we could report a more specific debug message about which specific upload failed.